src/anyio/_backends/_asyncio.py

```diff
@@ -304,6 +305,8 @@ def __exit__(self, exc_type: Optional[Type[BaseException]], exc_val: Optional[Ba
         if exc_val is not None:
             exceptions = exc_val.exceptions if isinstance(exc_val, ExceptionGroup) else [exc_val]
             if all(isinstance(exc, CancelledError) for exc in exceptions):
+                self._cancelled_caught = not self._parent_cancelled()
```

Comment:
I think cancel_caught should be false on native cancellation

Reply:
Indeed, and there should be a test for this.

Reply:
Not sure if I understood native cancellation correctly. I took it to mean "a scope that's been cancelled without calling .cancel()", which seems to only happen when `Cancelled` is thrown inside the scope's context manager.

I added a test to that effect. Let me know if I've missed something.